src/mlpack/core/metrics/bleu_impl.hpp

```diff
@@ -0,0 +1,178 @@
+/**
+ * @file core/metrics/bleu_impl.hpp
+ * @author Mrityunjay Tripathi
+ *
+ * Implementation of BLEU class.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_CORE_METRICS_BLEU_IMPL_HPP
+#define MLPACK_CORE_METRICS_BLEU_IMPL_HPP
+
+// In case it hasn't been included.
+#include "bleu.hpp"
+
+namespace mlpack {
+namespace metric {
+
+template <typename ElemType, typename PrecisionType>
+BLEU<ElemType, PrecisionType>::BLEU(const size_t maxOrder) :
+  maxOrder(maxOrder),
+  translationLength(0),
+  referenceLength(0)
```

Comment:
Tiny issue: according to the [style guide](https://github.com/mlpack/mlpack/wiki/DesignGuidelines#constructor-initialization-lists) you need 4 spaces here.